### PR TITLE
Add missing Batch Account test

### PR DESCRIPTION
**Pull request by matthchr**
**If applicable**:
- [x] this PR contains tests

### CODE REVIEW

File: v2/internal/controllers/crd_batch_account_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+func Test_Batch_Account_CRUD(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	// Custom namer because batch accounts have strict names
+	namer := tc.Namer.WithSeparator("")
```
Comment (theunrepentantgeek) [inline v2/internal/controllers/crd_batch_account_test.go:25]: This seems to be a really common requirement, perhaps we should add a helper to the test context?

File: v2/internal/controllers/crd_batch_account_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	}
+
+	tc.CreateResourceAndWait(account)
+
+	tc.Expect(account.Status.Id).ToNot(BeNil())
+	armId := *account.Status.Id
+
+	tc.DeleteResourceAndWait(account)
+
+	// Ensure that the cluster was really deleted in Azure
```
Comment (theunrepentantgeek) [inline v2/internal/controllers/crd_batch_account_test.go:42]:
```suggestion
	// Ensure that the account was really deleted in Azure
```

File: v2/internal/controllers/crd_batch_account_test.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	// Custom namer because batch accounts have strict names
+	namer := tc.Namer.WithSeparator("")
+
+	account := &batch.BatchAccount{
+		ObjectMeta: tc.MakeObjectMetaWithName(namer.GenerateName("batch")),
```
Comment (theunrepentantgeek) [inline v2/internal/controllers/crd_batch_account_test.go:28]:
```suggestion
		ObjectMeta: tc.MakeObjectMetaWithName(namer.GenerateName("batchacc")),
```

**Review by theunrepentantgeek — APPROVED** [review]

**Review by atlslscsrv-app — APPROVED** [review]

File: v2/internal/controllers/crd_batch_account_test.go (hunk 1 above)
Comment (matthchr) [reply v2/internal/controllers/crd_batch_account_test.go:25]: I'll do this but in a separate PR.

**Pull request merged**